uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/WorkPatternTest.php

```diff
@@ -70,44 +84,74 @@ public function testGetCalendarUsesTheWorkPatternCalendarService() {
     ]);
 
     WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => true]);
-    $multipleWeekWorkPattern = WorkPatternFabricator::fabricateWithTwoWeeksAnd31AndHalfHours();
 
-    // contract covers the whole period
+    // contracts covering the whole period
     HRJobContractFabricator::fabricate(
-      ['contact_id' => $contact['id']],
+      ['contact_id' => $contact1['id']],
       [
         'period_start_date' => '2015-01-05',
         'period_end_date' => '2015-01-19'
       ]
     );
 
-    // work pattern is effective only after the contract end
-    ContactWorkPatternFabricator::fabricate([
-      'contact_id' => $contact['id'],
-      'pattern_id' => $multipleWeekWorkPattern->id,
-      'effective_date' => CRM_Utils_Date::processDate('2015-02-05'),
-    ]);
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      [
+        'period_start_date' => '2015-01-05',
+        'period_end_date' => '2015-01-19'
+      ]
+    );
 
     $workDayTypes = $this->getWorkDayTypeOptionsArray();
 
-    $calendarDates = civicrm_api3('WorkPattern', 'getCalendar', [
+    $calendars = civicrm_api3('WorkPattern', 'getCalendar', [
       'period_id' => $absencePeriod->id,
-      'contact_id' => $contact['id']
+      'contact_id' => ['IN' => [$contact1['id'], $contact2['id']]]
     ])['values'];
 
+    $this->assertCount(2, $calendars);
+
+    // Since there's not guarantee about the order on which the calendars will
+    // be returned, we get all the contacts ID's from the result, put them in an
+    // array and then check if this array only contains the IDs we're expecting
+    $returnedContactsIDs = array_column($calendars, 'contact_id');
+    $this->assertContains($contact1['id'], $returnedContactsIDs);
+    $this->assertContains($contact2['id'], $returnedContactsIDs);
+
     // Asserting ALL the dates would make the test too big, so we assert the
     // total number of dates, the first and last ones and a few dates in the
     // middle, assuming all the other will be correct.
-    $this->assertCount(15, $calendarDates);
-    $this->assertEquals('2015-01-05', $calendarDates[0]['date']);
-    $this->assertEquals($workDayTypes['working_day'], $calendarDates[0]['type']);
-    $this->assertEquals('2015-01-08', $calendarDates[3]['date']);
-    $this->assertEquals($workDayTypes['working_day'], $calendarDates[3]['type']);
-    $this->assertEquals('2015-01-11', $calendarDates[6]['date']);
-    $this->assertEquals($workDayTypes['weekend'], $calendarDates[6]['type']);
-    $this->assertEquals('2015-01-16', $calendarDates[11]['date']);
-    $this->assertEquals($workDayTypes['working_day'], $calendarDates[11]['type']);
-    $this->assertEquals('2015-01-19', $calendarDates[14]['date']);
-    $this->assertEquals($workDayTypes['working_day'], $calendarDates[14]['type']);
+    // Since both contacts have the same work pattern, both calendars should be equal
+    foreach($calendars as $calendar) {
```

Comment:
Would be nice to test this with the two contacts using different work patterns. One contact can use the default work pattern and the other contact using a different work pattern.